com.avaloq.tools.ddk.check.ui/src/com/avaloq/tools/ddk/check/ui/builder/CheckEclipseGeneratorConfigProvider.java

```diff
@@ -35,11 +35,11 @@
   private CheckBuilderPreferenceAccess xbaseBuilderPreferenceAccess;
 
   @Override
-  public CheckGeneratorConfig get(final Resource resource) {
+  public CheckGeneratorConfig get(final URI uri) {
     CheckGeneratorConfig result = new CheckGeneratorConfig();
     IProject project = null;
-    if (resource != null) {
-      Pair<IStorage, IProject> pair = Iterables.getFirst(storage2UriMapper.getStorages(resource.getURI()), null);
+    if (uri != null) {

```

Comment:
I wonder why we had this null check before, as JavaDoc on the interface says that the parameter must not be null.

Reply:
do you want me to submit a new one changing the javadoc or removing the check?